### PR TITLE
[SDK-2374] Add Organizations support to Authentication API Client

### PR DESCRIPTION
**Pull request by evansims**
This PR adds support for Organizations (currently in a closed beta):

- `Auth0\SDK\Auth0` now supports an `organization` option in the constructor array. Forwards this to `Auth0\SDK\API\Authentication` automatically when used to create a singleton of the Authentication API class.
- `Auth0\SDK\API\Authentication` now supports an `organization` parameter in the constructor.
- `Auth0\SDK\API\Authentication::get_authorize_link()` will include the `?organization` parameter in the /authorize endpoint URL it builds, if configured in the parent class during instantiation or passed as a parameter (`$params`)  to the method.
- `Auth0\SDK\Auth0::login()` automatically inherits functionality for `?invitation` and `?organization` parameters from the parent class.

It also adds support for Invitations:

- Adds `handleInvitation()` method to `Auth0\SDK\Auth0` — This is a convenient method for triggering a redirect them invitation parameters are found in a GET request. If they aren't present, nothing happens. If they are, the end user's browser is redirected to kickstart the authorization.
- Adds `getInvitationParameters()` method to `Auth0\SDK\Auth0` — Used by `handleInvitation` but left `public` for customers who want more control over the invitation process; extracts the `invitation`, `organization`, and `organization_name` query parameters from a GET request and returns an object for working with them. If not present, returns null.

Also includes the following changes:
- Usage of Organizations and Invitations can be found in the updates to the README included.
- Adds helper functions `normalizeRequest` and `normalizeIncludedFields` for reducing code redundancy in building API requests, although these are not used directly in this PR (was intended for Management API calls involving Organizations, now moved to a separate PR, but I'd like to get these in immediately for general use)

Management API client work moved to separate PR: https://github.com/auth0/auth0-PHP/pull/483

### CODE REVIEW

File: src/API/Authentication.php [hunk 1]
```diff
@@ -52,6 +52,13 @@ class Authentication
      */
     private $audience;
 
+    /**
+     * Organization ID, if applicable.
```
Comment (stevehobbsdev) [inline src/API/Authentication.php:56]: Let's add a _tiny_ bit more context here. For example, elsewhere we've used:

> The Id of an organization to log in to

**Review by stevehobbsdev — CHANGES_REQUESTED** [review]
Generally looks good 🎉 added some minor content things.

~But also, I can see `organization` being promoted as a first-class param with code comments and such, we should also do the same for `invitation` as we have done elsewhere. Let me know if that doesn't make sense or I have misunderstood the code.~

Disregard ☝🏻 I can see why this isn't included given that there's a method to process a whole invitation URL. Carry on! 😄

File: README.md [hunk 2]
```diff
@@ -115,6 +123,94 @@ if (! empty($results)) {
...
+If you prefer to have more control over this process, a separate helper function is provided for extracting the query parameters, `getInvitationParameters()`, which you can use to initiate the Universal Login redirect yourself:
+
+```php
+// Expects the Auth0 SDK to be configured first, as demonstrated above.
+
+// Returns an object containing the invitation query parameters, or null if they aren't present
+if ($invite = $auth0->getInvitationParameters()) {
+  // Does the invite organization match your intended organization?
+  if ($invite->organization !== '{YOUR_ORGANIZATION_ID}') {
+    throw new Exception("These aren't the droids you're looking for.");
```
[the remark this answers is not in the document]
Comment (evansims) [reply README.md]: Cleaned that up!

File: src/API/Authentication.php (hunk 1 above)
Comment (evansims) [reply src/API/Authentication.php]: Good call, improved that with your example. Also expanded on src/Auth0.php's explenaton a bit as well.

**Comment by evansims** [comment]
@stevehobbsdev Thanks for the feedback, I applied those changes. Ready for you to have another look

**Review by stevehobbsdev — APPROVED** [review]
🎉

**Pull request merged**